src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import duke.exception.DukeFileException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Scanner;
+
+/**
+ * Encapsulates the storage class
+ */
+public class Storage {
+    private static SimpleDateFormat writeformatter = new SimpleDateFormat ("MMM dd yyyy hh:mm a");
+    private String filePath;
+
+    /**
+     * Creates a new storage
+     * @param filePath Path of the storage file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Writes the data of the TaskList into a file
+     * @param taskList TaskList to be written to file
+     * @return TaskList
+     * @throws DukeFileException If the file cannot be written to
+     */
+    public TaskList writeData(TaskList taskList) throws DukeFileException {
```

Comment:
Can seperate the TaskList from Storage class. Storage class might not need to know the TaskList class.